packages/services-context/src/index.ts

```diff
@@ -88,12 +90,32 @@ export class Services implements ServicesInterface {
 
   private async initializeWalletServices(): Promise<WalletServices> {
     const { walletId, fullViewingKey, idbVersion: dbVersion } = await this.config;
-    const params = await this.querier.app.appParams();
-    if (!params.sctParams?.epochDuration) throw new Error('Epoch duration unknown');
-    const {
-      chainId,
-      sctParams: { epochDuration },
-    } = params;
+
+    // read params from storage
+    const storedParams = await localExtStorage
+      .get('params')
+      .then(j => j && AppParameters.fromJson(j));
+    // try to fetch params from network
+    const queriedParams = await this.querier.app.appParams().catch((e: unknown) => void e);
+
+    // prefer fetched params
+    const chainId = queriedParams?.chainId ?? storedParams?.chainId;
+    const epochDuration =
+      queriedParams?.sctParams?.epochDuration ?? storedParams?.sctParams?.epochDuration;
+
+    // store chainId if it was missing
+    if (!storedParams?.chainId && queriedParams?.chainId)
+      await localExtStorage.set('params', queriedParams.toJson() as Jsonified<AppParameters>);
```

Comment:
Brackets here please

Reply:
i'm going to make a PR fixing the eslint config to reenable react-hooks, and i'll add a linter rule for braces at that point